python/lsst/cp/pipe/ptc.py

```diff
@@ -432,85 +485,141 @@ def funcAstier(self, pars, x):
         a00, gain, noise = pars
         return 0.5/(a00*gain*gain)*(np.exp(2*a00*x*gain)-1) + noise/(gain*gain)
 
-    def fitPtcAndNl(self, fitVectorsDict, ptcFitType='POLYNOMIAL'):
-        """Function to fit PTC, and calculate linearity and linearity residual
+    @staticmethod
+    def _initialParsForPolynomial(order):
+        assert(order >= 2)
+        pars = np.zeros(order, dtype=np.float)
+        pars[0] = 10
+        pars[1] = 1
+        pars[2:] = 0.0001
+        return pars
+
+    @staticmethod
+    def _boundsForPolynomial(initialPars):
```

Comment:
This function, as it is defined, seems to always return bounds that are independent on the argument "initialPars", namely [0, -inf, -inf, ..., -inf] and [inf, inf, ... , inf]. Something similar happens for the function below ("_boundsForAstier"). Is this intentional or maybe I am misinterpreting things?

Reply:
It takes `initialPars` as an argument because it needs to know how many there are, in order to return the right number of bounds (see it used in the list comps). The values are all plus/minus `inf` for the Astier fit, because they were previously unbounded, and I was just keeping them the same, whilst also adding both a place-holder method for adding bounds, should we want them in the future, whilst also not hard-coding the number of bounds into the code, especially as this is different for polynomial vs. Astier.